boopickle/src/main/scala/org/http4s/booPickle/BooPickleInstances.scala

```diff
@@ -0,0 +1,53 @@
+package org.http4s
+package booPickle
+
+import boopickle.Default._
+import boopickle.Pickler
+import cats.Applicative
+import cats.effect.Sync
+import fs2.Chunk
+import java.nio.ByteBuffer
+import org.http4s._
+import org.http4s.EntityEncoder.chunkEncoder
+import org.http4s.headers.`Content-Type`
+import scala.util.{Failure, Success}
+
+/**
+  * Generic factories for http4s encoders/decoders for boopickle
+  * Note that the media type is set for application/octet-stream
+  */
+trait BooPickleInstances {
+
+  private def booDecoderByteBuffer[F[_]: Sync, A](msg: Message[F])(
+      implicit pickler: Pickler[A]): DecodeResult[F, A] =
+    EntityDecoder.collectBinary(msg).flatMap { segment =>
+      val bb = ByteBuffer.wrap(segment.force.toArray)
+      if (bb.hasRemaining) {
+        Unpickle[A](pickler).tryFromBytes(bb) match {
+          case Success(bb) =>
+            DecodeResult.success[F, A](bb)
+          case Failure(pf) =>
+            DecodeResult.failure[F, A](MalformedMessageBodyFailure("Invalid binary body", Some(pf)))
+        }
+      } else {
+        DecodeResult.failure[F, A](MalformedMessageBodyFailure("Invalid binary: empty body", None))
+      }
+    }
+
+  /**
+    * Create an `EntityDecoder` for `A` given a `Pickler[A]`
+    */
+  def booOf[F[_]: Sync, A: Pickler]: EntityDecoder[F, A] =
+    EntityDecoder.decodeBy(MediaType.`application/octet-stream`)(booDecoderByteBuffer[F, A])
```

Comment:
This is going to be hard to use, like `jsonOf` became hard to use in 0.18, isn't it?

Reply:
I’m not sure what you mean, could you elaborate?

Reply:
I see. I noted `decodeJson` was on `Message` but I assumed it was grandfathered. OTOH I don't think you want to keep adding `decodeXXX` methods. Or perhaps they  can be extension methods in the bundle?